gitops/snapshot.go

```diff
@@ -703,3 +706,25 @@ func RemoveIntegrationTestRerunLabel(adapterClient client.Client, ctx context.Co
 
 	return nil
 }
+
+func GetLatestUpdateTime(snapshot *applicationapiv1alpha1.Snapshot) (time.Time, error) {
+	latestUpdateTime := snapshot.GetAnnotations()[SnapshotPRLastUpdate]
+	if latestUpdateTime == "" {
+		return time.Time{}, nil
+	}
+	t := time.Time{}
+	err := t.UnmarshalText([]byte(latestUpdateTime))
+	if err != nil {
+		return t, fmt.Errorf("failed to unmarshal time annotation: %w", err)
+	}
+	return t, nil
+}
+
+func SetLatestUpdateTime(snapshot *applicationapiv1alpha1.Snapshot, t time.Time) error {
```

Comment:
Same here, I think we'll also need some dedicated unit tests for this to have enough code coverage.